File: android-reference-app/app/src/main/java/io/agora/android_reference_app/AuthenticationActivity.java
package io.agora.android_reference_app;

import io.agora.agora_manager.AgoraManager;
import io.agora.authentication_manager.AuthenticationManager;

import android.os.Bundle;
import android.view.SurfaceView;
import android.widget.EditText;

public class AuthenticationActivity extends BasicImplementationActivity {
    private AuthenticationManager authenticationManager;
    //private Button btnJoinLeave;
    private EditText editChannelName; // To read the channel name from the UI.
    private EditText editServerUrl; // To read the server Url from the UI.

    @Override
    protected int getLayoutResourceId() {
        return R.layout.activity_authentication; // Default layout resource ID for base activity
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        editChannelName = findViewById(R.id.editChannelName);
        editChannelName.setText(agoraManager.channelName);

        editServerUrl = findViewById(R.id.editServerUrl);
        editServerUrl.setText(authenticationManager.serverUrl);
    }

    @Override
    protected void initializeAgoraManager() {
        authenticationManager = new AuthenticationManager(this);
        agoraManager = authenticationManager;

        // Set the current product depending on your application
        agoraManager.setCurrentProduct(AgoraManager.ProductName.VIDEO_CALLING);
        // Set up a listener for updating the UI
        agoraManager.setListener(agoraManagerListener);
    }

    @Override
    protected void join() {
        String channelName = editChannelName.getText().toString();
        authenticationManager.serverUrl = editServerUrl.getText().toString();
        authenticationManager.fetchToken(channelName, new AuthenticationManager.TokenCallback() {
            @Override
            public void onTokenReceived(String rtcToken) {
                // Handle the received rtcToken
                agoraManager.joinChannel(channelName, rtcToken);
                showLocalVideo();
            }

            @Override
            public void onError(String errorMessage) {
                // Handle the error
                System.out.println("Error: " + errorMessage);
            }
        });
        btnJoinLeave.setText(R.string.leave);
    }
}

File: android-reference-app/app/src/main/java/io/agora/android_reference_app/AuthenticationActivity.kt
package io.agora.android_reference_app

import io.agora.authentication_manager.AuthenticationManager
import io.agora.authentication_manager.AuthenticationManager.TokenCallback

import android.widget.EditText
import android.os.Bundle

class AuthenticationActivity : BasicImplementationActivity() {
    private lateinit var authenticationManager: AuthenticationManager
    private var editChannelName // To read the channel name from the UI.
            : EditText? = null
    private var editServerUrl // To read the server Url from the UI.
            : EditText? = null

    // Override the UI layout
    override val layoutResourceId: Int
        get() = R.layout.activity_authentication

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Set up access to the UI elements
        editChannelName = findViewById(R.id.editChannelName)
        editChannelName!!.setText(agoraManager.channelName)
        editServerUrl = findViewById(R.id.editServerUrl)
        editServerUrl!!.setText(authenticationManager.serverUrl)
    }

    override fun initializeAgoraManager() {
        // Instantiate an object of the AuthenticationManager class, which is an extension of the AgoraManager
        authenticationManager = AuthenticationManager(this)
        agoraManager = authenticationManager

        // Set up a listener for updating the UI
        agoraManager.setListener(agoraManagerListener)
    }

    override fun join() {
        // Read the channel name
        val channelName = editChannelName!!.text.toString()
        // Read the authentication server URL
        authenticationManager.serverUrl = editServerUrl!!.text.toString()
        if (authenticationManager.serverUrl == "") {
            showMessage("You must specify an token server URL")
            return
        }

        // Fetch a token from the server using an async call
        authenticationManager.fetchToken(channelName, object : TokenCallback {
            override fun onTokenReceived(rtcToken: String?) {
                // Join a channel using the token
                agoraManager.joinChannel(channelName, rtcToken)
            }

            override fun onError(errorMessage: String) {
                // Fetch token failed
                println("Error: $errorMessage")
            }
        })
    }
}